patch/cisco-x86-gpio-config.patch

```diff
@@ -0,0 +1,42 @@
+From 2ee30f75a96d5f7841ffd58cd65410160302421b Mon Sep 17 00:00:00 2001
+From: Madhava Reddy Siddareddygari <msiddare@cisco.com>
+Date: Tue, 13 Jul 2021 18:43:16 -0700
+Subject: [PATCH] Configurable GPIO option ARCH_NR_GPIO for x86 architecture
+
+The x86 platform did not allow configuring the maximum number of GPIOs
+supported, although the ARM platform did. For cisco-8000 platform,
+each FPGA gpio IP block can support 1K pins. Distributed chassis with
+RP and FC can have 10 such IP blocks, along with additional pins through
+i2c gpio extenders.
+
+This patch supports configurable number of GPIO's at kernel config time
+similar to ARM platform.
+
+Signed-off-by: Madhava Reddy Siddareddygari <msiddare@cisco.com>
+---
+ arch/x86/Kconfig | 9 +++++++++
+ 1 file changed, 9 insertions(+)
+
+diff --git a/arch/x86/Kconfig b/arch/x86/Kconfig
+index d2453b251..21804d1f0 100644
+--- a/arch/x86/Kconfig
++++ b/arch/x86/Kconfig
+@@ -1033,6 +1033,15 @@ config SCHED_MC_PRIO
+ 
+ 	  If unsure say Y here.
+ 
++config ARCH_NR_GPIO
++	int
++	prompt "Maximum number of GPIO's"
++	default 512
++	---help---
```

Comment:
Why the dashes?

Reply:
Most of the config options [ Not all ] have convention of adding dashes to the "help".
Only very few do not have dashes.  

Patch updated to remove dashes